src/data/list/basic.lean

```diff
@@ -3480,6 +3480,60 @@ begin
     { simp [hmem] } }
 end
 
+@[simp] lemma enum_nil : enum ([] : list α) = [] := rfl
+@[simp] lemma enum_from_nil (n : ℕ) : enum_from n ([] : list α) = [] := rfl
+
+lemma enum_from_cons (x : α) (xs : list α) (n : ℕ) :
+  enum_from n (x :: xs) = (n, x) :: enum_from (n + 1) xs := rfl
+lemma enum_cons (x : α) (xs : list α) :
+  enum (x :: xs) = (0, x) :: enum_from 1 xs := enum_from_cons _ _ _
```

Comment:
```suggestion
  enum (x :: xs) = (0, x) :: enum_from 1 xs := rfl
```

Reply:
BTW, should these two lemmas be `simp`? If yes, then `simp` will compute `enum [a, b, c, d]`.

Reply:
Happy to make the change. A lot of the other `enum` lemmas are not tagged with `simp`, so I followed that.